components/scream/src/share/util/scream_common_physics_functions.hpp

```diff
@@ -351,15 +351,26 @@ struct PhysicsFunctions
   // Note: because this function does an integral it cannot be run just on a single level.  It requires
   // the full column wise integration.
   //-----------------------------------------------------------------------------------------------//
-  template<typename ScalarT, typename InputProviderZ>
+  template<typename InputProviderZ, typename ViewT>
```

Comment:
Templating on the view type is fine, but as a reference, another valid approach is to do
```
template<typename ScalarT, typename InputProviderZ, typename MT = Kokkos::MemoryManaged>
static void calculate_z_int (...., const view_1d<ScalarT, MT>& z_int);
```
since `view_1d` in KokkosTypes supports also an optional memory traits template parameter (which defaults to MemoryManaged).